plugin.video.southpark_unofficial/southpark.py

```diff
@@ -171,26 +159,51 @@ def __init__(self, addon):
 		self.OPTIONS_GEOLOCATION                  = _translation(addon, 30001)
 		self.OPTIONS_PLAY_DIRECTLY_RANDOM_EPISODE = _translation(addon, 30015)
 		self.WARNING_BAD_INTERNET_CONNECTION      = _translation(addon, 30010)
-		self.WARNING_BANNED_EPISODE               = _translation(addon, 30011)
+		self.WARNING_UNAVAILABLE_EPISODE          = _translation(addon, 30011)
 		self.WARNING_GEOBLOCKED                   = _translation(addon, 30002)
 		self.WARNING_LOADING                      = _translation(addon, 30009)
 		self.WARNING_LOADING_RANDOM_EPISODE       = _translation(addon, 30003)
 		self.WARNING_PREMIERE                     = _translation(addon, 30014)
 
+
+class KodiParams(object):
+	"""Kodi parameters"""
+	def __init__(self, param_string):
+		super(KodiParams, self).__init__()
+		params = _decode_dictionary(param_string)
+		self.PARAM_MODE    = unquote_plus(params.get('mode'     , ''))
+		self.PARAM_SEASON  = unquote_plus(params.get('season'   , ''))
+		self.PARAM_EPISODE = unquote_plus(params.get('episode'  , ''))
+
+	def debug(self):
+		log_debug("PARAM_MODE:    {0}".format(self.PARAM_MODE   ))
+		log_debug("PARAM_SEASON:  {0}".format(self.PARAM_SEASON ))
+		log_debug("PARAM_EPISODE: {0}".format(self.PARAM_EPISODE))
+
+class SP_Paths(object):
+	"""South Park plugin Paths"""
+	def __init__(self, addon_id):
+		super(SP_Paths, self).__init__()
+		self.PLUGIN_ICON      = xbmc.translatePath('special://home/addons/{0}/icon.png'.format(addon_id))
```

Comment:
please use xbmcvfs.translatePath for matrix and onwards. You can use `IS_PY3` to distinguish calls since py3 is matrix++ only

Reply:
what is the import ? `import xbmcvfs` ?

Reply:
yes